flink-container/src/main/java/org/apache/flink/container/entrypoint/ClassPathJobGraphRetriever.java

```diff
@@ -110,19 +117,38 @@ public JobGraph retrieveJobGraph(Configuration configuration) throws FlinkExcept
 		}
 	}
 
-	private PackagedProgram createPackagedProgram() throws FlinkException {
-		final String entryClass = getJobClassNameOrScanClassPath();
+	private PackagedProgram createPackagedProgram(final String entryClass, final boolean isPythonProgram) throws FlinkException {
 		try {
 			final Class<?> mainClass = getClass().getClassLoader().loadClass(entryClass);
-			return new PackagedProgram(mainClass, programArguments);
-		} catch (ClassNotFoundException | ProgramInvocationException e) {
+			return new PackagedProgram(mainClass, getProgramArguments(isPythonProgram));
+		} catch (ProgramInvocationException e) {
 			throw new FlinkException("Could not load the provided entrypoint class.", e);
+		} catch (ClassNotFoundException e) {
+			if (!isPythonProgram && jobPythonArtifacts != null) {
+				return createPackagedProgram("org.apache.flink.python.client.PythonDriver", true);
```

Comment:
This is overly complicated and error prone. Why not starting the cluster entrypoint with `o.a.f.python.client.PythonDriver` as `jobClassName`?